core/block/editor/clipboard/clipboard.go

```diff
@@ -564,32 +565,54 @@ func (cb *clipboard) addRelationLinksToDataview(d *model.BlockContentDataview) (
 
 func renderText(s *state.State) string {
 	texts := make([]string, 0)
-	childrenIds := s.Pick(s.RootId()).Model().ChildrenIds
-	for _, id := range childrenIds {
-		texts = renderChildren(s, texts, id, 0)
-	}
+	texts, _ = renderBlock(s, texts, s.RootId(), -1, 0)
 
 	if len(texts) > 0 {
-		return strings.Join(texts, "\n")
+		return strutil.JoinWithTrailingEnd(texts, "\n")
 	}
 
 	return ""
 }
 
-func renderChildren(s *state.State, texts []string, id string, level int) []string {
+func renderBlock(s *state.State, texts []string, id string, level int, numberedCount int) ([]string, int) {
 	block := s.Pick(id).Model()
-	if text := block.GetText(); text != nil {
-		if level > 0 {
-			texts = append(texts, strings.Repeat("\t", level)+text.Text)
-		} else {
-			texts = append(texts, text.Text)
+	texts, numberedCount = extractTextWithStyleAndTabs(block, texts, level, numberedCount)
+	childrenIds := s.Pick(id).Model().ChildrenIds
+	texts = renderChildren(s, texts, childrenIds, level, 0)
+	return texts, numberedCount
+}
+
+func renderChildren(s *state.State, texts []string, childrenIds []string, level int, numberedCount int) []string {
+	var oldNumberedCount int
+	for _, id := range childrenIds {
+		oldNumberedCount = numberedCount
+		texts, numberedCount = renderBlock(s, texts, id, level+1, numberedCount)
+		if oldNumberedCount == numberedCount {
```

Comment:
Is it for splitting numbered list for separate chunks? Like

```
# One
# Two
Text
# One
```
renders to
```
1. One
2. Two
Text
1. One
```

Reply:
yep